### PR TITLE
chore: update gitops-engine to c036d3f6b0e2

### PR DESCRIPTION
**Pull request by leoluz**
Signed-off-by: Leonardo Luz Almeida <leonardo_almeida@intuit.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,7 +9,7 @@ require (
 	github.com/TomOnTime/utfutil v0.0.0-20180511104225-09c41003ee1d
 	github.com/alicebob/miniredis v2.5.0+incompatible
 	github.com/alicebob/miniredis/v2 v2.14.2
-	github.com/argoproj/gitops-engine v0.7.1-0.20220804190909-2bc3fef13e07
+	github.com/argoproj/gitops-engine v0.7.1-0.20220819195816-9970faba8138
```
Comment (ashutosh16) [inline go.mod:12]: I think you need to include gitops change https://github.com/argoproj/gitops-engine/commit/a56a8030319ee3af0f6287865133403f8f9ad380 to fix integration test.

Comment (leoluz) [reply go.mod:12]: @ashutosh16 actually it is the opposite. I updated gitops-engine with current master which already includes that commit. The problem is that, the signature change in `gitopsengine.sync.WithNamespaceCreation` breaks current ArgoCD master. Basically current gitops-engine in master is incompatible with current ArgoCD in master.

@pasha-codefresh @jannfis What would be the best course of action in this case? Maybe we should think about how we can prevent this from happening in the future? WDYT?

**Review by ashutosh16 — APPROVED** [review]
lgtm! as  all test passing

**Review by jannfis — APPROVED** [review]
LGTM

**Pull request merged**